scripts/check-submodules.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+set -eou pipefail
+
+function escape {
+  v="$1"
+  v="${v//%/%25}"
+  v="${v//$'\n'/%0A}"
+  v="${v//$'\r'/%0D}"
+  echo -n "$v"
+}
+
+function log {
+  msg_type="$1"
+  title="$2"
+  if [[ -t 0 ]]; then
+    case "$msg_type" in
+    "error")
+      echo -ne '\033[0;31m'
+      ;;
+    "warning")
+      echo -ne '\033[0;33m'
+      ;;
+    "notice")
```

Comment:
It seems like the "notice" level logs also go to stderr. Is this intended behaviour?

Reply:
Yes, it's intended.  I think it's a human-readable diagnostic message, and the script's only output for machine is just the exit code.  (This SO link might be worth to read: <https://unix.stackexchange.com/a/331620/9939>.)

Reply:
Ooh! Thanks for the pointer. Glad to learn that :)